src/commands/lock.rs

```diff
@@ -0,0 +1,108 @@
+//! Implementation of the `lock` subcommand.
+
+use std::collections::HashMap;
+use std::collections::HashSet;
+
+use anyhow::Result;
+use anyhow::bail;
+use clap::Parser;
+use crankshaft_docker::Docker as crankshaft_docker;
+use serde::Deserialize;
+use serde::Serialize;
+use wdl::ast::AstToken;
+use wdl::ast::v1::Expr;
+use wdl::ast::v1::LiteralExpr;
+use wdl::cli::Analysis;
+use wdl::cli::analysis::Source;
+
+/// Default name for the lock file.
+const LOCK_FILE: &str = "sprocket.lock";
+
+/// Arguments for the `lock` subcommand.
+#[derive(Parser, Debug)]
+pub struct Args {
+    /// A source WDL file or URL.
+    #[clap(value_name = "PATH or URL")]
+    pub source: Source,
```

Comment:
Don't love this API; I would change it to one of these two UXs:
1. no arguments, `sprocket lock` recurses the CWD finding all WDL files and locking each image
2. A single argument for `workspace` that _must_ be a directory. `sprocket.lock` gets written to `<workspace>/`

Either way, we need to operate on directories instead of individual files.... well should you be able to "lock" a remote workflow? 🤔 now I'm not sure....

Reply:
Ah interesting. Well as written that's not how it works. As written, you are only locking the tasks in the supplied file, as opposed to the tasks _called by the workflow_

What you describe here should probably also fail if there's no workflow in the supplied source WDL